src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,48 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
```

Comment:
Very good. Making it abstract can help to ensure no "Contact" object can be created.